File: 프로그래머스/0/120837. 개미 군단/개미 군단.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int hp) {
    int answer = 0;
    int a[3] = {5, 3, 1};
    int i = 0;
    while(hp > 0 && i < 3) {
        if(hp >= a[i]) hp -= a[i], answer++;
        else i++;
    }
    return answer;
}

File: 프로그래머스/0/120837. 개미 군단/개미 군단.swift
import Foundation

func solution(_ hp:Int) -> Int {
    // return (hp / 5) + (hp % 5 / 3) + (hp % 5 % 3)
    
    var result = 0
    var hp = hp
    for i in [5, 3, 1] {
        result += hp / i
        hp %= i
    }
    return result
}
